### PR TITLE
DAOS-13070 control: Port daos cont set-prop handler to Go

### PR DESCRIPTION
**Pull request by kjacque**
Removed the redundant C handler and updated the Go code to call daos_cont_set_prop directly.

Features: daos_cmd

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/cmd/daos/container.go [hunk 1]
```diff
@@ -1297,20 +1297,24 @@ func (cmd *containerSetPropCmd) Execute(args []string) error {
 	}
 	defer deallocCmdArgs()
 
-	ap.props = cmd.Args.Props.props
```
Comment (mjmac) [inline src/control/cmd/daos/container.go:1300]: I think that you'll need to add a `defer cmd.Args.Props.Cleanup()` here in order to make NLT happy.

File: src/control/cmd/daos/container.go [hunk 2]
```diff
@@ -1297,20 +1297,24 @@ func (cmd *containerSetPropCmd) Execute(args []string) error {
...
+	rc := C.daos_cont_set_prop(ap.cont, cmd.Args.Props.props, nil)
 	if err := daosError(rc); err != nil {
-		return errors.Errorf("failed to set properties on container %s",
+		return errors.Wrapf(err, "failed to set properties on container %s",
 			cmd.ContainerID())
 	}
 
+	if cmd.jsonOutputEnabled() {
+		return cmd.outputJSON(nil, nil)
+	}
```
Comment (mjmac) [inline src/control/cmd/daos/container.go:1314]: I don't _think_ this is necessary... Have you tried it without? The way it should work is that if `--json` was supplied, the `cmd.Info` output is suppressed, so the end result should be the same as if you call `cmd.outputJSON(nil, nil)`. If it's not, that's a bug IMO.

File: src/control/cmd/daos/container.go [hunk 3]
```diff
@@ -1297,20 +1297,24 @@ func (cmd *containerSetPropCmd) Execute(args []string) error {
...
+
+	cmd.Info("Properties were successfully set\n")
```
Comment (mjmac) [inline src/control/cmd/daos/container.go:1316]: No newlines needed with our logging.

**Review by mjmac — COMMENTED** [review]
Nice, thanks for getting this last handler. I noticed it when I was working on the other ones but for some reason thought there was more to it. Should have switched it over ages ago...

File: src/control/cmd/daos/container.go (hunk 1 above)
Comment (kjacque) [reply src/control/cmd/daos/container.go:1300]: Good catch. I guess it was previously getting cleaned up as part of the `ap` struct.

File: src/control/cmd/daos/container.go (hunk 3 above)
Comment (kjacque) [reply src/control/cmd/daos/container.go:1316]: Fixed.

**Pull request merged**